src/api/services/companyService.ts

```diff
@@ -222,3 +223,50 @@ export function addCalculatedTotalEmissions(companies: any[]) {
       }))
   )
 }
+
+export function addCompanyEmissionChange(companies: any[]) {
+  return( companies.map((company) => {
+    company.reportingPeriods.sort((a, b) => new Date(b.endDate).getTime() - new Date(a.endDate).getTime());
+    company.reportingPeriods.map((period: any, index: number) => {
+      if(index < company.reportingPeriods.length - 1) {
+        let adjustedCurrentTotal = 0;
+        let adjustedPreviousTotal = 0;
+        const previousPeriod = company.reportingPeriods[index + 1];
+        const { scope1: currentScope1, scope2: currentScope2, scope3: currentScope3 } = period.emissions || {};
+        const { scope1: preivousScope1, scope2: preivousScope2, scope3: preivousScope3 } = previousPeriod.emissions || {};
+        if(currentScope1 && preivousScope1) {
+          adjustedCurrentTotal += currentScope1?.total ?? 0;
+          adjustedPreviousTotal += preivousScope1?.total ?? 0;
+        }
+        if(currentScope2 && preivousScope2) {
+          adjustedCurrentTotal += currentScope2?.mb ?? currentScope2?.lb ?? currentScope2?.unknown ?? 0;
+          adjustedPreviousTotal += preivousScope2?.mb ?? preivousScope2?.lb ?? preivousScope2?.unknown ?? 0;
+        }
+        if(currentScope3 && preivousScope3) {
+          if(currentScope3?.categories && preivousScope3?.categories && currentScope3?.categories.length > 0 && preivousScope3?.categories.length > 0) {
+            currentScope3?.categories.forEach((currentCategory) => {
+              const previousCategory = preivousScope3?.categories.find((category) => category.category === currentCategory.category);
+              if(previousCategory) {
+                adjustedCurrentTotal += currentCategory?.total ?? 0;
+                adjustedPreviousTotal += previousCategory?.total ?? 0;
+              }
+            });
+          } else if(currentScope3.statedTotalEmissions && preivousScope3.statedTotalEmissions) {
+            adjustedCurrentTotal += currentScope3?.statedTotalEmissions ?? 0;
+            adjustedPreviousTotal += preivousScope3?.statedTotalEmissions ?? 0;
+          }
+        }
+        period.emissionsTrend = {
+          absolute: period.emissions.calculatedTotalEmissions > 0 ? ((period.emissions.calculatedTotalEmissions - previousPeriod.emissions.calculatedTotalEmissions) / previousPeriod.emissions.calculatedTotalEmissions * 100) : 0,
+          adjusted: adjustedCurrentTotal > 0 ? ((adjustedCurrentTotal - adjustedPreviousTotal) / adjustedPreviousTotal * 100) : 0
+        }
+      } else {
+        period.emissionsTrend = {
+          absolute: null,
+          adjusted: null
+        }
+      }
+    });
+    return company;
```

Comment:
Would be awesome to split this function up a bit to make it easier to read and test.